TMT Mac/Views/MoviePosterView.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  MoviePosterView.swift
+//  TheatricalMovieTrailers
+//
+//  Created by Christoph Parstorfer on 15.10.20.
+//  Ported to macOS by Christoph Parstorfer on 24.12.21.
+//
+
+import SwiftUI
+
+struct MoviePosterView: View {
+    private let filmPosterAspectRatio = CGFloat(0.7063020214)
+    
+    @State private var reflectionDistance: CGFloat
+    @State private var onTap: (() -> ())?
+    @Binding private var image: NSImage?
+    @State private var blurReflection = true
+    
+    init(image: Binding<NSImage?>, reflectionDistance: CGFloat = -1.0, blurReflection: Bool = true, onTapGesture: (() -> ())? = nil) {
+        self._reflectionDistance = State<CGFloat>(initialValue: reflectionDistance)
+        self._blurReflection = State<Bool>(initialValue: blurReflection)
+        self._onTap = State<(() -> ())?>(initialValue: onTapGesture)
+        self._image = image
+    }
+    
+    var body: some View {
+        var image: NSImage!
```

Comment:
I use the NSImage class because I need to mask the image with the gradient image at runtime. I had initially attempted to use SwiftUI's `.mask` modifier to achieve this, but it did not work.
https://github.com/conath/TheatricalMovieTrailers/blob/f38864db0da5b219ec5aa024125a33dc58e1b57d/TMT%20Mac/Views/MoviePosterView.swift#L34

I just attempted to do this again and it didn't work. I changed to a transparent-to-white gradient image mask and applied it to the reflection image. It seems like transparent images are not treated as having a less than 1 alpha value in SwiftUI, because this code did not work:

```swift
        return GeometryReader { geo in
            HStack {
                Spacer()
                VStack(alignment: .center, spacing: reflectionDistance) {
                    Image(nsImage: image)
                        .resizable()
                        .aspectRatio(contentMode: .fit)
                        .frame(maxHeight: geo.size.height * 2 / 3)
                    Image(nsImage: image)
                        .resizable()
                        .aspectRatio(contentMode: .fit)
                        .mask(Image("posterMirrorImageGradientAlphaMask"))
                        .opacity(0.7)
                        .frame(maxHeight: geo.size.height * 2 / 3)
                        .rotationEffect(.degrees(180), anchor: .center)
                        .rotation3DEffect(
                            .degrees(180),
                            axis: (x: 0, y: 1, z: 0)
                        )
                        .blur(radius: blurReflection ? 1 : 0)
    // ...
```

However I did get it working by using the original image (gradient black to white) and applying the `.luminanceToAlpha()` modifier to the mask view:

```swift
    .mask(
        Image("posterMirrorImageGradientMask")
            .resizable()
            .aspectRatio(contentMode: .fit)
            .frame(maxHeight: geo.size.height / 2) // placeholder for testing
            .luminanceToAlpha()
    )
```
🏆